src/vsphere_cpi/spec/unit/cloud/vsphere/network_spec.rb

```diff
@@ -324,14 +334,19 @@ module VSphereCloud
         end
 
         context 'if more than 1 attached' do
-          let(:switch_ports) { [instance_double(NSXT::LogicalPort), instance_double(NSXT::LogicalPort)] }
+          let(:switch_ports) { [logical_port, logical_port] }
           it 'raises an error' do
             expect(router_provider).to receive(:get_attached_router_ids)
               .with('switch-id').and_return(['t1-router-id'])
             expect(switch_provider).to receive(:get_attached_switch_ports)
               .with('switch-id').and_return(switch_ports)
             expect(switch_provider).to receive(:get_switch_by_id)
              .with('switch-id').and_return(logical_switch)
+
+            expect(logical_port).to receive(:id)
+             .twice.and_return("12345")
```

Comment:
This expectation can also go away then. I don't think logical port getting this twice is adding any validity to the test. We can permanently allow this by adding as I commented above.

other thoughts?